### ISSUE TITLE
X-COM UFO Defense Initial Base Setup Option

### ISSUE THREAD

**Issue by Nurulinde** [OpenApoc/OpenApoc#1207]
In UFO Defense, the player starts with an empty base, and a number of facilities which can be placed anywhere in the base the player desires.  It would be nice to have the option for this kind of start in OpenApoc as well.  I understand there are some purists who like the RNG-like pre-built bases which might offer an additional logistical challenge, I am only suggesting this as an option.  It could be simply implemented by refunding the cost of the Bases initial facilities into the Players Bank... then allowing the player to "Build/Place" the individual allowable/unlocked facilities wherever desired.. then once completed, Set all the "days to completion" timers to zero (completed).

**Comment by FilmBoy84** [comment]
I agree, this would be a good QoL feature

However, a number of problems present themselves

* Special changes would have to be made to the gamestate on new-game to allow the loading of the base facility construction screen before the city and parent building are loaded - this would mean that RNG for the building selection in the chosen citymap would have to be performed before the map itself is loaded, then the user presented with a means to place each standard facility, then continuing to load the starting gamestate after that

* The existing base facility screen includes buttons and options that would not be available (or functional) until the game has fully loaded and started - it's highly likely we'd need a new screen using assets from the Original Game as appropriate. A solution that does this must NOT distribute Original Game assets with OpenApoc for legal reasons, so any use of Original Game assets must be constructed from the copy that the end user of OpenApoc provides themselves

* Special consideration needs to be taken with regards to the advantages of allowing this and the impact on game balance - allowing users to select their own layout on game start provides an distinct advantage to early base defence and no financial penalty for moving stuff around that would normally be applicable - I suggest the starting base is still loaded as per RNG would currently decide, then the player can rearrange the base - at cost - to retain the impact on starting finances. This doesn't get around the advantage of being able to place stuff where you want before the first risk of attack, but at least you are still taking a financial hit for it

**Comment by Nurulinde** [comment]
I suppose one problem might be, until all the "Living Quarters" are built/commited, not sure what the game would do with the initial Agent and Scientist allotment.

**Comment by Nurulinde** [comment]
Unless, of course, in this mode the Agents and Scientists were de-populated too... Bank account increased a bit more... and the player would just hire new ones after building the base.  That might work

**Comment by Artytoo** [comment]
Look I went here in '02.

And yes like UFO and TFTD it was good to place base entry points in a linear path having to go through defense stations and saving you lots of agents and lost damages.

I remember there being a graphical glitch with the outside of the base when it was tried and it would skew and blur graphics.

+ Yes I want a complete Cityscape "underground" with Xcom Base graphics to get around in. "I mention this on FB"

In all the decades of playing it seemed sappers were your best shot.
- Basically saved game editing.

If sappers were implemented as a mod I could see plenty of people using it as a savegame editor for maps.
- Maybe what we need is sappers, and or Map editors.
